src/bm_sim/lpm_trie.h

```diff
@@ -21,32 +21,106 @@
 #ifndef BM_SIM_LPM_TRIE_H_
 #define BM_SIM_LPM_TRIE_H_
 
-#include <bf_lpm_trie/bf_lpm_trie.h>
+#include <bm/bm_sim/bytecontainer.h>
 
 #include <algorithm>  // for std::swap
-#include <utility>  // for std::swap
+#include <utility>   // for std::swap
+#include <string>
+#include <memory>
+#include <cassert>
+#include <cstdint>
+#include <vector>
 
 namespace bm {
 
-static_assert(sizeof(value_t) == sizeof(uintptr_t),
-              "Invalid type sizes");
+// Hao: value_t is a c type, equivalent to unsigned long, but need to make
+// sure
+typedef std::uintptr_t value_t;
+typedef unsigned char byte_t;
```

Comment:
I'd prefer not to have these at the top-level. You can define the type aliases you need as part of the `LPMTrie` scope ideally.